content/handling-multiple-k8sclusters.md

```diff
@@ -0,0 +1,73 @@
+
+In case you ended up with multiple configs in one file, and you want to separate them, you can extract all information for one specific context with
+
+    kubectl config view --minify --flatten --context=my_context > myconfig.yaml. 
+
+## Auto-populate $KUBECONFIG
+
+how to set up $KUBECONFIG so that it automatically picks up all files from ~/.kube/
+
+If the set of clusters you interact with changes often, one elegant way to deal with that is to populate your KUBECONFIG variable automatically. There's many ways to do this, but here's what I'm using. I have this line in my environment:
+
+    set -xg KUBECONFIG $HOME/.kube/config:(ls -m $HOME/.kube/*.yaml | tr ',' ':' | tr -d "[:blank:]") 
+
+ Note that I'm using fish as my default shell, hence I'm using fish's native `set` command with the `-x` flag instead of `export`. In a normal bash or zsh environment you would use `export KUBECONFIG=...` for the same purpose. 
+
+This picks up `~/.kube/config as well as all files with a .yaml file extension from the ~/.kube directory, and stores them as colon separated list in my KUBECONFIG variable. So if I add a new cluster, I just have to put the config there and re-source my environment. 
+
+## Useful tools and tricks
+
+So far we've obviously only covered Kubernetes' built-in functionality. Which is basically fine, but the are a couple of little helpers and additional tricks that are worth knowing about. 
+
+The first tool that I want to call out is Ahmet Balkan's kubectx and its brother-in-arms, kubens. Both are available from the same [GitHub repo](https://github.com/ahmetb). There's really not much to explain about them - they give you a shorter way to switch between contexts and namespaces than their somewhat lengthy kubectl counterparts, plus they basically behave like the cd command for changing directories (e.g. `kubectx -` gets you back to the previous context). Nothing magic, but neat. Check the embedded videos in the GitHub readme, and you'll get the idea. 
+
+Another trick many Kubernetes people use is to display current cluster and namespace your command line prompt. You might think this is unnecessary, but wait until you accidentally delete a pod from the wrong namespace. If you are using bash or zsh (the default shell on MacOS), you can simply use [https://github.com/jonmosco/kube-ps1](https://github.com/jonmosco/kube-ps1). You can install it via your favorite package manager, and you'll have to add a few lines to your .bashrc or .zshrc. The process is described well in the repository's readme. If you're using [fish](https://fishshell.com), you may want to take a look at [https://github.com/aluxian/fish-kube-prompt] for the same purpose (tip: have a look at [fisher](https://github.com/jorgebucaran/fisher) to manage your fish extensions). Both prompt mods can be turned on and off easily with `kubeon|kubeoff` for kube-ps1 and `kube_ps on|off` for fish-kube-prompt. 
```

Comment:
```suggestion
Another trick many Kubernetes people use is to display current cluster and namespace your command line prompt. You might think this is unnecessary, but wait until you accidentally delete a pod from the wrong namespace. If you are using bash or zsh (the default shell on MacOS), you can simply use [https://github.com/jonmosco/kube-ps1](https://github.com/jonmosco/kube-ps1). You can install it via your favorite package manager -- you'll have to add a few lines to your .bashrc or .zshrc. The repository's readme describes the process well. If you're using [fish](https://fishshell.com), take a look at [https://github.com/aluxian/fish-kube-prompt] for the same purpose (tip: have a look at [fisher](https://github.com/jorgebucaran/fisher) to manage your fish extensions). You can easily turn both prompt mods on and off with `kubeon|kubeoff` for kube-ps1 and `kube_ps on|off` for fish-kube-prompt. 
```

Reply:
Took this on board in my incoming commit.